vllm/v1/engine/__init__.py

```diff
@@ -90,7 +90,13 @@ class EngineCoreOutputs(
     # e.g. columnwise layout
 
     # [num_reqs]
-    outputs: List[EngineCoreOutput]
+    request_ids: List[str]
+    new_token_id_offsets: List[int]
+    new_token_ids: List[int]
+    # TODO: need offsets for logprobs?
+    new_logprobs: List[Optional[LogprobsLists]]
+    new_prompt_logprobs_tensors: List[Optional[LogprobsTensors]]
+    finish_reason: Dict[str, FinishReason]  # Union[List, Dict]?
```

Comment:
Could make this `Dict[str, Tuple[FinishReason, Union[int, str, None]]` where the tuple is (finish_reason, stop_reason)

Reply:
afaict, the stop reason isn't used anywhere anymore so I got rid of it. Are there plans to use it in the future?

Reply:
It is used .. it's part of the API i.e. information provided back to the user in the final response along with the finish reason.

Reply:
It's passed into `self._make_request_output(...` just below that, and set within the returned `RequestOutput` in the finished case.

Reply:
Fixed in #13108